src/main/scala/net/ripe/rpki/publicationserver/PublicationService.scala

```diff
@@ -0,0 +1,122 @@
+package net.ripe.rpki.publicationserver
+
+import java.io.ByteArrayInputStream
+
+import akka.actor._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.`Content-Type`
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.http.scaladsl.util.FastFuture
+import akka.pattern.ask
+import akka.util.Timeout
+import com.softwaremill.macwire._
+import javax.xml.stream.XMLStreamException
+import net.ripe.rpki.publicationserver.messaging.Messages.RawMessage
+import net.ripe.rpki.publicationserver.model.ClientId
+import net.ripe.rpki.publicationserver.parsing.PublicationMessageParser
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.io.{BufferedSource, Source}
+import scala.util.{Failure, Success}
+import akka.util.ByteString
+
+object PublicationService {
+    val MediaTypeString = "application/rpki-publication"
+    val `rpki-publication` = MediaType.customWithFixedCharset("application", "rpki-publication", HttpCharsets.`UTF-8`)
+} 
+
+class PublicationService
+    (conf: AppConfig, stateActor: ActorRef)
+    (implicit val system: ActorSystem) extends Logging {
+
+  implicit val executionContext = system.dispatcher
+
+  val supervisorStrategy = OneForOneStrategy(maxNrOfRetries = 1) {
+    case _: Exception =>
+      SupervisorStrategy.Escalate
+  }
+  
+  val msgParser = wire[PublicationMessageParser]
+
+  // TODO Support proper streaming
+  implicit val BufferedSourceUnmarshaller: Unmarshaller[HttpEntity, BufferedSource] =
+    Unmarshaller.withMaterializer { _ => 
+        implicit mat => {
+            case HttpEntity.Strict(_, data) => 
+                FastFuture.successful(Source.fromInputStream(new ByteArrayInputStream(data.toArray)))
+            case entity                     => 
+                entity.dataBytes
+                    .runFold(ByteString.empty)(_ ++ _)
+                    .map(data => Source.fromInputStream(new ByteArrayInputStream(data.toArray)))
```

Comment:
Could it be `Source.fromBytes(data.toArray)`, to simplify things?

Reply:
That won't compile, because fromBytes returns Source and we need BufferedSource :)
So everything around would have to e changed to Source, including xml parser, etc.

Reply:
Uuh!